tools/dynamic_framework_slicer/dynamic_framework_slicer.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash 
+set -euo pipefail
+
+declare -a BINARIES
+IN=""
+OUT=""
+while [[ $# -gt 0 ]]; do
+  arg="$1"
+  case $arg in
+      --in)
+      readonly IN="$2"
+      shift
+      ;;
+      --out)
+      readonly OUT="$2"
+      shift
+      ;;
+      *)
+      BINARIES+=("$1")
+      ;;
+  esac
+  shift
+done
+
+if [[ -z "$IN" || -z "$OUT" || -z "${BINARIES[*]}" ]]; then
```

Comment:
actually, the array might need to check the count is >0, i think this expansion will be multiple args otherwise and the test will be invalid.

Reply:
Actually multiple args was okay, it was not enough args that failed poorly. It is better now.